### PR TITLE
added mongoose model for a User

### LINKED ISSUE #7: Create mongoose model for user
https://medium.com/better-programming/how-to-use-mongoose-with-node-js-913a8073b29c
https://medium.com/@tomanagle/strongly-typed-models-with-mongoose-and-typescript-7bc2f7197722

**start this only when task #3 is completed**

### PR DESCRIPTION
**Pull request by Sienkowski99**
closes #7

### CODE REVIEW

File: src/models/User.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { Schema, model } = require('mongoose');
```
Comment (bartlomiejmont) [inline src/models/User.ts]: use import

Comment (Sienkowski99) [reply src/models/User.ts]: It's fixed. I'm not sure if what i've added is enough but it should work.

File: src/models/User.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import * as mongoose from "mongoose";
+
+const UserSchema = new mongoose.Schema({
+  login: { type: String, required: true, unique: true },
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true, unique: true },
```
Comment (bartlomiejmont) [inline src/models/User.ts]: are passwords really unique ?

File: src/models/User.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  login: { type: String, required: true, unique: true },
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true, unique: true },
+  registrationDate: Date,
+  lastLoggedIn: Date,
+});
+
+const UserModel = mongoose.model("User", UserSchema);
+
+export {UserModel};
```
Comment (bartlomiejmont) [inline src/models/User.ts]: We can write `export const UserModel = mongoose.model("User", UserSchema);
` instead

Comment (bartlomiejmont) [inline src/models/User.ts]: Add User Interface https://stackoverflow.com/questions/34482136/mongoose-the-typescript-way

File: src/models/User.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+const UserSchema = new mongoose.Schema({
+  login: { type: String, required: true, unique: true },
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true, unique: true },
+  registrationDate: Date,
+  lastLoggedIn: Date,
+});
+
+const UserModel = mongoose.model("User", UserSchema);
```
Comment (bartlomiejmont) [inline src/models/User.ts]: name this User insted of UserModel

**Review by bartlomiejmont — CHANGES_REQUESTED** [review]

File: src/models/User.ts (hunk 4 above)
Comment (Sienkowski99) [reply src/models/User.ts]: fixed

File: src/models/User.ts (hunk 3 above)
Comment (Sienkowski99) [reply src/models/User.ts]: done

Comment (Sienkowski99) [reply src/models/User.ts]: changed

File: src/models/User.ts (hunk 2 above)
Comment (Sienkowski99) [reply src/models/User.ts]: silly copy and paste mistake - fixed

**Review by bartlomiejmont — APPROVED** [review]

**Pull request merged**